src/Functions/greatCircleDistance.cpp

```diff
@@ -295,13 +295,13 @@ class FunctionGeoDistance : public IFunction
         return dst;
     }
 
-    const ColumnFloat32 * convertArgumentColumnToFloat32(const ColumnsWithTypeAndName & arguments, size_t argument_index) const
+    const ColumnFloat64 * convertArgumentColumnToFloat64(const ColumnsWithTypeAndName & arguments, size_t argument_index) const
     {
-        const auto * column_typed = checkAndGetColumn<ColumnFloat32>(arguments[argument_index].column.get());
+        const auto * column_typed = checkAndGetColumn<ColumnFloat64>(arguments[argument_index].column.get());
         if (!column_typed)
             throw Exception(
                     ErrorCodes::ILLEGAL_COLUMN,
-                    "Illegal type {} of argument {} of function {}. Must be Float32.",
+                    "Illegal type {} of argument {} of function {}. Must be Float64.",
```

Comment:
Can we allow both Float32 and Float64?

Reply:
The function signature is of ColumnFloat64 though.

Reply:
The logic should be as follows:
- if at least one argument is Float64, it uses the Float64 specialization;
- if all arguments are Float32, it uses the Float32 specialization.